app/src/ui/history/compare.tsx

```diff
@@ -0,0 +1,466 @@
+    const placeholderText =
+      selectedBranch === null
+        ? __DARWIN__
+          ? 'Select Branch To Compare...'
+          : 'Select branch to compare...'
+        : undefined
+
+    return (
+      <div id="compare-view">
+        <div className="the-box">
+          <FancyTextBox
+            symbol={OcticonSymbol.gitBranch}
+            placeholder={placeholderText}
+            onFocus={this.onTextBoxFocused}
+            onBlur={this.onTextBoxBlurred}
+            value={this.state.filterText}
+            onRef={this.onTextBoxRef}
+            onValueChanged={this.onBranchFilterTextChanged}
+            onKeyDown={this.onBranchFilterKeyDown}
+          />
+        </div>
+        {showBranchList ? this.renderFilterList() : this.renderCommits()}
+      </div>
+    )
+  }
+
+  private renderCommits() {
+    const compareState = this.props.repositoryState.compareState
+
+    return (
+      <div className="the-commits">
+        {compareState.kind === CompareType.None
+          ? this.renderCommitList()
+          : this.renderTabBar()}
+      </div>
+    )
+  }
+
+  private renderCommitList() {
+    const compareState = this.props.repositoryState.compareState
+    const selectedCommit = this.state.selectedCommit
+    const commitSHAs = compareState.commitSHAs
+
+    return (
+      <CommitList
+        gitHubRepository={this.props.repository.gitHubRepository}
+        commitLookup={this.props.commitLookup}
+        commitSHAs={commitSHAs}
+        selectedSHA={selectedCommit !== null ? selectedCommit.sha : null}
+        gitHubUsers={this.props.gitHubUsers}
+        localCommitSHAs={this.props.localCommitSHAs}
+        emoji={this.props.emoji}
+        onViewCommitOnGitHub={this.props.onViewCommitOnGitHub}
+        onRevertCommit={this.props.onRevertCommit}
+        onCommitSelected={this.onCommitSelected}
+        onScroll={this.onScroll}
+      />
+    )
+  }
+
+  private renderActiveTab() {
+    if (this.state.selectedTab === SelectedTab.Behind) {
+      return (
+        <div className="the-commits">
+          {this.renderCommitList()}
+          {this.renderMergeCTA()}
+        </div>
+      )
+    } else {
+      return <div className="the-commits">{this.renderCommitList()}</div>
+    }
+  }
+
+  private renderFilterList() {
+    const {
+      branches,
+      recentBranches,
+      defaultBranch,
+      currentBranch,
+    } = this.branchState
+
+    return (
+      <BranchList
+        defaultBranch={defaultBranch}
+        currentBranch={currentBranch}
+        allBranches={branches}
+        recentBranches={recentBranches}
+        filterText={this.state.filterText}
+        textbox={this.textbox!}
+        selectedBranch={this.state.selectedBranch}
+        canCreateNewBranch={false}
+        onSelectionChanged={this.onSelectionChanged}
+        onFilterTextChanged={this.onBranchFilterTextChanged}
+        renderBranch={this.renderCompareBranchListItem}
+      />
+    )
+  }
+
+  private renderMergeCTAMessage() {
```

Comment:
`CTA` isn't an acronym I'm familiar with, and I might be found out to be foolish for asking but I'm gonna do it anyway:

 - what does it stand for?
 - can something more descriptive be used here?

Reply:
Probably call-to-action, but I’m not positive.

Reply:
Yep, what @j-f1 said.

Reply:
We have a few mentions of `CallToAction` already in the codebase, so my gut feeling is that we could simplify this down to `renderMergeSummary` and name the below function `renderMergeCallToAction` rather than abbreviate it especially here. I'll add this into #4390 while I'm cleaning up things.